controllers/addons_controller.go

```diff
@@ -115,10 +127,12 @@ func (r *AddonsLayerReconciler) processPrune(l layers.Layer) (statusReconciled b
 
 	pruneIsRequired, hrs, err := applier.PruneIsRequired(ctx, l)
 	if err != nil {
+		r.Log.Error(err, "check for apply required failed", "requestName", l.GetName())
 		return false, err
 	} else if pruneIsRequired {
 		l.SetStatusPruning()
-		if pruneErr := applier.Prune(ctx, l, hrs); err != nil {
+		if pruneErr := applier.Prune(ctx, l, hrs); pruneErr != nil {
+			r.Log.Error(pruneErr, "prune failed", "requestName", l.GetName())
```

Comment:
If it's necessary to log an error with additional context, it probably makes sense to wrap the original error to add that context before returning it to the caller.

Reply:
sorry not following what you are saying here